src/ai/backend/common/plugin/__init__.py

```diff
@@ -127,16 +128,39 @@ def __init__(self, etcd: AsyncEtcd, local_config: Mapping[str, Any]) -> None:
     def discover_plugins(
         cls,
         plugin_group: str,
+        allowlist: set[str] = None,
         blocklist: set[str] = None,
     ) -> Iterator[Tuple[str, Type[P]]]:
+        combined_allowlist: Optional[set] = None
+        if cls.allowlist is not None:
+            if combined_allowlist is None:
+                combined_allowlist = set()
+            combined_allowlist |= cls.allowlist
+        if allowlist is not None:
+            if combined_allowlist is None:
+                combined_allowlist = set()
+            combined_allowlist |= allowlist
         cls_blocklist = set() if cls.blocklist is None else cls.blocklist
         arg_blocklist = set() if blocklist is None else blocklist
-        for entrypoint in scan_entrypoints(plugin_group, cls_blocklist | arg_blocklist):
+        for entrypoint in scan_entrypoints(
+            plugin_group,
+            combined_allowlist,
```

Comment:
Could we just refactor this line like `cls_allowlist | arg_allowlist`? I think it's possible to reduce the newly added codes in this method into about 5 lines.